9_collaborator_data/src/psu_site_selection.R

```diff
@@ -315,33 +315,47 @@ plot_upstream <- function(POIs, dist_mat_ind, labels=c('subseg_id','seg_id_nat')
 
 }
 
+# identify those subsegs whose outlets are in the geospatial fabric (omitting
+# reaches we added for the purpose of constructing and accurate distance matrix)
+select_sntemp_subsegs <- function(
+  network_ind = '1_network/out/network.rds.ind',
+  out_rds) {
 
-subset_dist <- function(dat, dist_ind, network_ind, out_ind){
-
-  sites <- readRDS(dat)
-  dist_mat <- readRDS(sc_retrieve(dist_ind))
-  dist_mat <- dist_mat$updown
   network <- readRDS(sc_retrieve(network_ind))
+  sntemp_edges <- network$edges %>%
+    mutate(end_subseg=sprintf('%dd', subseg_seg)) %>%
+    filter(
+      mapply(function(ep, ss) {
+        grepl(sprintf('%dd', ss), ep)
+      }, ep=end_pt, ss=subseg_seg)) %>%
+    select(subseg_id, seg_id_nat)
+
+  saveRDS(sntemp_edges, out_rds)
+}
 
-  subset_dist <- dist_mat[sites$subseg_id, ]
-  subset_dist <- subset_dist[, sites$subseg_id]
+subset_dist_to_subsegs <- function(subsegs_rds, dist_ind, network_ind, out_ind){
 
-  saveRDS(subset_dist, as_data_file(out_ind))
+  subsegs <- readRDS(subsegs_rds)
+  network <- readRDS(sc_retrieve(network_ind))
+  dist_mats <- readRDS(sc_retrieve(dist_ind))
+  subset_dists <- lapply(dist_mats, function(dist_mat) {
```

Comment:
question: what is `dist_mat`? I don't see it defined anywhere.

Reply:
oh nvm. it's a temporary function you are defining here.

Reply:
jk again 😆 . It's the argument to the function you are defining here - the variable represents the members of the list